### ISSUE TITLE
Declare language for user localized text

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-frontend#3049]
### Describe the problem and steps to reproduce it:

* Browse the site in your default locale (such as French)
* Go an add-on detail page where the default locale of the description is something other than French (say, English)

### What happened?

The section of the text in English does not have a `lang` attribute which means it will be interpreted as French (or whatever locale is in the URL).

### What did you expect to happen?

It should have a `lang` attribute specifying the locale.

### Anything else we should know?

* This is important for [screen readers](http://adrianroselli.com/2015/01/on-use-of-lang-attribute.html).
* The old site did it.
* We should probably also include the `dir` attribute for right-to-left languages but I don't know what the practical usage of that is.
* This should apply to all user localized data, such as reviews, version release notes, etc.

**Comment by muffinresearch** [comment]
I mentioned this case to @eviljeff and he pointed out that if you specify lang in an API request you'll get only the locale you asked for. This however will fall back to other locales e.g. if you request `lang=fr` some descriptions will be in french but others will be in English but there's no data to be able to know that a given description is not the locale you requested. 

To fix this we will need a fairly extensive breaking change in the API to annotate all localized content with the locale of the actual data returned to support this case.

**Comment by eviljeff** [comment]
For reference http://addons-server.readthedocs.io/en/latest/topics/api/overview.html#api-overview-translations is what we do at the moment.  

We could change the API to always return a dict (with only the requested locale, or the default locale otherwise).  But that would be a breaking change as mentioned (I don't know how reliant frontend is on the existing behavior).  Or add another param that would return the dict, or some other structure that indicates the locale of each property.

@diox and @EnTeQuAk to look and ponder|despair.

**Comment by kumar303** [comment]
Ah, right, I see

**Comment by diox** [comment]
> We could change the API to always return a dict (with only the requested locale, or the default locale otherwise)

That would be my preference, because more query parameters are awful, but... backwards-compatibility breakage :( Still, the frontend could be modified in advance to cope with this change maybe? The APIs are marked as experimental in the docs precisely to allow us to do things like that, we don't guarantee their stability to the outside world.

**Comment by eviljeff** [comment]
Okay, so essentially I think we're agreed on always returning a dict for localised values.  As its a breaking change frontend will need to do the work beforehand to handle it, or at least have the patch ready to merge.  I don't think the API change is massive so shouldn't be much work on the addons-server side (though there will be many, many, test changes)

**Comment by eviljeff** [comment]
fwiw, this isn't blocked on API changes (and hasn't been for a few years) - the API response always indicates the locale of any localised field now.

**Comment by ioanarusiczki** [comment]
@GregSherman 

Should name also have the lang attribute? 

![name](https://github.com/mozilla/addons-frontend/assets/33448286/a27c31f0-ca63-4bb2-870a-45575a4107fc)



 There's a lang attribute for summary, description, developer comments and e-mail:

![summary](https://github.com/mozilla/addons-frontend/assets/33448286/0b5f2188-46ef-440d-9181-04d4e7729c2c)
![description](https://github.com/mozilla/addons-frontend/assets/33448286/2ac9c4e3-cf6f-4f07-8301-3f860ada0cc3)
![developer comments](https://github.com/mozilla/addons-frontend/assets/33448286/4ef369d0-dd5c-4e04-a1cf-730afa098ae2)
![email](https://github.com/mozilla/addons-frontend/assets/33448286/66f79a62-c786-4eba-a730-f4fdc4cd7964)

**Comment by diox** [comment]
This likely caused https://github.com/mozilla/addons-frontend/issues/12227

**Comment by eviljeff** [comment]
this was reverted, so reopening.

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDFRNT-119
